ref/general/authorization.adoc

```diff
@@ -15,17 +15,18 @@
 
 In order for an application to be secure, it must be able to manage access to protected resources. Authorization determines which resources a user can access.
 
-In an application where only certain users or groups are entitled to access certain resources, users and groups are assigned roles that represent which resources they can access. For example, in a payroll application, users in the `Employee` role can view a page that lists all employees and their contact information. However, only users in the `Manager` role can view information about how much each employee is paid. During link:/docs/ref/general/#authentication.html[authentication], Open Liberty determines who a user is and what roles they are in. During authorization, Open Liberty determines how these roles map to different protected resources on the server.
+In an application where only certain users or groups are entitled to access certain resources, users and groups are assigned roles that represent which resources they can access. For example, in a payroll application, users in the `Employee` role can view a page that lists their hourly wages and hours worked. However, only users in the `Manager` role can make changes to employee wages. During link:/docs/ref/general/#authentication.html[authentication], Open Liberty determines who a user is and what roles they are in. During authorization, Open Liberty determines how these roles correspond to different protected resources on the server.
+
+The roles that users and groups are checked against can either be defined directly in the application or applied to the application by way of the server configuration. It is also possible to avoid defining roles altogether if you specify role names that match the group names in the user registry exactly. In this case, the server  maps the role to the group with a matching name by default. In this configuration, you do not specify an `<application-bnd>` element in the `server.xml` file.
```

Comment:
this is still not right. The roles are defined in the application, what can be specified in configuration is the mapping from a role to a user or group. The way this is written it makes it sound like the role itself is defined in server.xml. I would also not say to define the role names to match the groups, but the opposite way around. The role is the primary artifact.

Reply:
rewrote this section top clarify, lmk if it works.

I changed the orientation between roles vs groups in this section to give roles precedence- but I'm confused. The default mapping occurs when roles match groups- but dont groups, esp as defined in an LDAP registry or by OIDC provider exist independent of any one application? Wouldn't an app dev have to know what user groups were defined in the existing security framework and then define roles to match?

Reply:
There is a difference between how Role Based Access Control works from a design perspective, and how it is implemented from a performance perspective and I think the docs are written from an implementation perspective. 

A role is a collection of permissions, so a permission could allow you to read or write a file, or access a servlet, or an ejb. A role collects a bunch of disparate permissions together so you can manage them as a collection more easily. Then you "map" users or groups as having that role. The idea is some bit of code says "does the user have permission X" and at some point there is a mapping from a user and the groups the user is in to the roles those users and groups have and to the permissions. This can all be processing and memory intensive, so internally what tends to happen is the security check just says "does the user have this role" and we store the role data with the rest of the identity data. However this is an internal optimization and not really how we should describe RBAC.

From this perspective I think this is still worrying me:

> During link:/docs/ref/general/#authentication.html[authentication], Open Liberty determines who a user is and what roles they are in. During authorization, Open Liberty determines how these roles correspond to different protected resources on the server.

I think it probably should read like this:

> During link:/docs/ref/general/#authentication.html[authentication], Open Liberty identifies the user, what groups the user might be in, and which roles they have. When accessing a protected resource on the server this information is used to determine if the user is authorized to access the protected resource. 

For the second paragraph I think we need to simplify:

>The roles ~that users and groups are checked against~ are defined directly in the application. Out of the box the role names are mapped directly to group names in the user registry or single sign-on token. Often group names do not directly map to role names as defined in the application. In this case a mapping from role names to users and groups can be defined in server.xml.

Reply:
@NottyCode pushed a new commit to make these changes